akka-http-core/src/test/scala/akka/http/impl/util/AkkaSpecWithMaterializer.scala

```diff
@@ -25,7 +34,20 @@ abstract class AkkaSpecWithMaterializer(s: String)
       // main system guardian being shutdown which will be after the logging has
       // reverted to stdout logging that cannot be intercepted
       materializer.asInstanceOf[ActorMaterializer].shutdown()
+      Http().shutdownAllConnectionPools()
       // materializer shutdown is async but cannot be watched
       Thread.sleep(10)
     }
 }
+object AkkaSpecWithMaterializer {
+  // adapted version of AkkaSpec.getCallerName that also works for `AkkaSpecWithMaterializer`
+  def getCallerName(clazz: Class[_]): String = {
+    val s = (Thread.currentThread.getStackTrace map (_.getClassName) drop 1)
```

Comment:
@jrudolph is in the house!

Reply:
Wasn't my idea but just copied from AkkaSpec ;) I'm not even quite sure why it's needed, somehow it seemed that due to the initializers or something, the concrete class wasn't available otherwise.

Reply:
This came back after all and needed another fix: [https://github.com/akka/akka-http/pull/3218/commits/458807442d5a88d219297f7f54b35b7d78667fda ](https://github.com/akka/akka-http/pull/3218/commits/458807442d5a88d219297f7f54b35b7d78667fda) ;)